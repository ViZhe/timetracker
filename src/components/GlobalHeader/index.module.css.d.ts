
export const header: string;
export const trigger: string;
