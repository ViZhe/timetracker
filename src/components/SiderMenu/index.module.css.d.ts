
export const sider: string;
export const logo: string;
export const logoIcon: string;
export const logoName: string;
