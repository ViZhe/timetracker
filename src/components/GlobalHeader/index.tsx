
import React, { Component } from 'react';
import { Layout, Icon } from 'antd';

import styles from './index.module.css';


type Props = {
  collapsed: boolean;
  onClickTrigger: () => void;
};

const { Header } = Layout;

class GlobalHeader extends Component<Props> {
  render() {
    const { collapsed, onClickTrigger } = this.props;
    return (
      <Header className={styles.header}>
        <Icon
          className={styles.trigger}
          type={collapsed ? 'menu-unfold' : 'menu-fold'}
          onClick={onClickTrigger}
        />
      </Header>
    );
  }
}


export default GlobalHeader;
