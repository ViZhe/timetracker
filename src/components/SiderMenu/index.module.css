
.sider {
  position: relative;
  z-index: 10;
  min-height: 100vh;
  box-shadow: 2px 0 6px rgba(0, 21, 41, 0.35);
}

.logo {
  overflow: hidden;
  display: flex;
  align-items: center;
  height: 32px;
  margin: 16px;
  padding-left: .5rem;
}

.logoIcon {
  font-size: 32px;
  color: #fff;
}

.logoName {
  margin-left: 1rem;
  color: #fff;
  font-size: 20px;
  transition: all .25s ease;
}

:global(.ant-layout-sider-collapsed) .logoName {
  margin-left: .5rem;
  opacity: 0;
}
