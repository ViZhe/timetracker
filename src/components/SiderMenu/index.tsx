
import { Icon, Layout, Menu } from 'antd';
import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import styles from './index.module.css';


interface IProps {
  collapsed: boolean;
  onCollapse: () => void;
}

const { Sider } = Layout;

class SiderMenu extends Component<IProps, {}> {
  render() {
    const { collapsed, onCollapse }: IProps = this.props;

    return (
      <Sider
        className={styles.sider}
        collapsible={true}
        collapsed={collapsed}
        onCollapse={onCollapse}
      >
        <Link to="/">
          <div className={styles.logo}>
            <Icon type="clock-circle" className={styles.logoIcon} />
            <span className={styles.logoName}>TimeTracker</span>
          </div>
        </Link>
        <Menu theme="dark" mode="inline" defaultSelectedKeys={['time']}>
          <Menu.Item key="time">
            <Link to="/time">
              <Icon type="home" />
              <span>Home</span>
            </Link>
          </Menu.Item>
          <Menu.Item key="reports">
            <Link to="/reports">
              <Icon type="pie-chart" />
              <span>Reports</span>
            </Link>
          </Menu.Item>
          <Menu.Item key="settings">
            <Link to="/settings">
              <Icon type="setting" />
              <span>Settings</span>
            </Link>
          </Menu.Item>
          <Menu.Item key="notmatch">
            <Link to="/notmatch">
              <Icon type="lock" />
              <span>Error</span>
            </Link>
          </Menu.Item>
        </Menu>
      </Sider>
    );
  }
}


export default SiderMenu;
