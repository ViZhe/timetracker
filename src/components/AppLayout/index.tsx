
import { Layout } from 'antd';
import React, { useState } from 'react';

import GlobalHeader from '../../components/GlobalHeader';
import SiderMenu from '../../components/SiderMenu';


const AppLayout: React.FC = ({ children }) => {
  const [isCollapsed, setCollapsed] = useState<boolean>(false);

  const toggleCollapsed = () => setCollapsed(!isCollapsed);

  return (
    <Layout>
      <SiderMenu collapsed={isCollapsed} onCollapse={toggleCollapsed} />
      <Layout>
        <GlobalHeader collapsed={isCollapsed} onClickTrigger={toggleCollapsed} />
        {children}
      </Layout>
    </Layout>
  );
};


export default AppLayout;
