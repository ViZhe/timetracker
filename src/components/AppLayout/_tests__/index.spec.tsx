
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import GlobalHeader from '../../../components/GlobalHeader';
import SiderMenu from '../../../components/SiderMenu';
import AppLayout from '../../AppLayout';


describe('<AppLayout />', () => {
  describe('<AppLayout /> initial', () => {
    const appLayout: ShallowWrapper = shallow(<AppLayout />);

    it('should match snapshot', () => {
      expect(appLayout).toMatchSnapshot();
    });

    it('should be not collapsed by default', () => {
      expect(appLayout.find(SiderMenu).prop('collapsed')).toBeFalsy();
      expect(appLayout.find(GlobalHeader).prop('collapsed')).toBeFalsy();
    });
  });

  describe('<AppLayout /> children', () => {
    const appLayout: ShallowWrapper = shallow(<AppLayout><div>test</div></AppLayout>);

    it('should match snapshot', () => {
      expect(appLayout).toMatchSnapshot();
    });

    it('should have children', () => {
      expect(appLayout.find('div').text()).toEqual('test');
    });
  });
});
