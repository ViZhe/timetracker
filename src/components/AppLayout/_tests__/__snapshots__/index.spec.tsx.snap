// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLayout /> <AppLayout /> children should match snapshot 1`] = `
<Adapter>
  <Component
    collapsed={false}
    onCollapse={[Function]}
  />
  <Adapter>
    <Component
      collapsed={false}
      onClickTrigger={[Function]}
    />
    <div>
      test
    </div>
  </Adapter>
</Adapter>
`;

exports[`<AppLayout /> <AppLayout /> initial should match snapshot 1`] = `
<Adapter>
  <Component
    collapsed={false}
    onCollapse={[Function]}
  />
  <Adapter>
    <Component
      collapsed={false}
      onClickTrigger={[Function]}
    />
  </Adapter>
</Adapter>
`;
